src/transaction.js

```diff
@@ -586,53 +585,51 @@ class Transaction {
     addRekey(reKeyTo, feePerByte=0) {
         if (reKeyTo !== undefined) {
-            this.reKeyTo = address.decode(reKeyTo);
+            this.reKeyTo = address.decodeAddress(reKeyTo);
         }
         if (feePerByte !== 0) {
             this.fee += (ALGORAND_TRANSACTION_REKEY_LABEL_LENGTH + ALGORAND_TRANSACTION_ADDRESS_LENGTH) * feePerByte;
         }
     }
 }
 
-
 /**
- * Aux class for group id calculation of a group of transactions
+ * encodeUnsignedTransaction takes a completed txnBuilder.Transaction object, such as from the makeFoo
+ * family of transactions, and converts it to a Buffer
+ * @param transactionObject the completed Transaction object
+ * @returns {Uint8Array}
  */
-class TxGroup {
-    constructor(hashes) {
-        if (hashes.length > ALGORAND_MAX_TX_GROUP_SIZE) {
-            let errorMsg = hashes.length.toString() + " transactions grouped together but max group size is " + ALGORAND_MAX_TX_GROUP_SIZE.toString();
-            throw Error(errorMsg);
-        }
-
-        this.name = "Transaction group";
-        this.tag = Buffer.from("TG");
-
-        this.txGroupHashes = hashes;
-    }
-
-    get_obj_for_encoding() {
-        const txgroup = {
-            "txlist": this.txGroupHashes
-        };
-        return txgroup;
-    }
-
-    static from_obj_for_encoding(txgroupForEnc) {
-        const txn = Object.create(this.prototype);
-        txn.name = "Transaction group";
-        txn.tag = Buffer.from("TG");
-        txn.txGroupHashes = [];
-        for (let hash of txgroupForEnc.txlist) {
-            txn.txGroupHashes.push(new Buffer.from(hash));
-        }
-        return txn;
-    }
+function encodeUnsignedTransaction(transactionObject) {
+    let objToEncode = transactionObject.get_obj_for_encoding();
+    return encoding.encode(objToEncode);
+}
 
-    toByte() {
-        return encoding.encode(this.get_obj_for_encoding());
-    }
+/**
+ * decodeUnsignedTransaction takes a Buffer (as if from encodeUnsignedTransaction) and converts it to a txnBuilder.Transaction object
+ * @param transactionBuffer the Uint8Array containing a transaction
+ * @returns {Transaction}
+ */
+function decodeUnsignedTransaction(transactionBuffer) {
+    let partlyDecodedObject = encoding.decode(transactionBuffer);
+    return Transaction.from_obj_for_encoding(partlyDecodedObject);
+}
 
+/**
+ * decodeSignedTransaction takes a Buffer (from transaction.signTxn) and converts it to an object
+ * containing the Transaction (txn), the signature (sig), and the auth-addr field if applicable (sgnr)
+ * @param transactionBuffer the Uint8Array containing a transaction
+ * @returns {Object} containing a Transaction, the signature, and possibly an auth-addr field
+ */
+function decodeSignedTransaction(transactionBuffer) {
+    let stxnDecoded = encoding.decode(transactionBuffer);
+    stxnDecoded.txn = Transaction.from_obj_for_encoding(stxnDecoded.txn);
+    return stxnDecoded;
 }
 
-module.exports = {Transaction, TxGroup, ALGORAND_MIN_TX_FEE};
```

Comment:
Similar to my above reply, I believe changing where `TxGroup` is defined should not break any supported uses of the SDK since it only changes internal files.

Re: exporting functions, I will change it so that `TxGroup` is not publicly exported, since it is only used by our mocha tests, which can import internal files. However `Transaction` is used by cucumber steps, which can't import internal files because it runs against our bundled browser code. If it's important that this class does not get publicly exported, I can rewrite the two steps that use it.